src/lucky/router.cr

```diff
@@ -1,19 +1,32 @@
 class Lucky::Router
-  INSTANCE = new
+  alias RouteMatch = LuckyRouter::Match(Lucky::Action.class) | Lucky::FallbackRoute | Nil
```

Comment:
Instead of having multiple types here, what if we remove `Lucky::FallbackRoute` and instead instantiate a `LuckyRouter::Match`. Something like this:

```crystal
  def find_action(method, path) : RouteMatch
     @matcher.match(method.to_s.downcase, path) || LuckyRouter::Match.new(fallback, {} of String => String)
end
```

And that or is pretty long so might be worth extracting a private def `fallback_match`

Reply:
I do like this alias though and think it'd be cool to keep it!

Reply:
Wouldn't that require updating that shard separately?

Reply:
You can instantiate a `LuckyRouter::Match` by yourself since it is a public class. As long as the first method match the signature (`Lucky::Action.class` in this case) then it will return the same object. I believe what I wrote should work, but it requires the fallback to be a class not an instance (so it matches the type signature)